### ISSUE TITLE
`stylistic.configs.customize({ quotes: "double" })` allows unnecessary usage of backticks

### ISSUE THREAD

**Issue by ssalggnikool** [eslint-stylistic/eslint-stylistic#1248]
### Validations

- [x] Follow our [Code of Conduct](https://github.com/eslint-stylistic/eslint-stylistic/blob/main/CODE_OF_CONDUCT.md)
- [x] Read the [Contributing Guide](https://eslint.style/contribute/guide).
- [x] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.

### Describe the bug

Using `stylistic.configs.customize({ quotes: "double" })` still lets you use backticks, even if you aren't using string interpolation/template literals.

See `index.ts` and `eslint.config.js` in the reproduction I linked. If you run `lint` task there will be no errors despite `quotes` being set to `double`.

**This issue does not occur if you manually set the rule** with `"@stylistic/quotes": ["error", "double"]`, only if you use `customise`

### Reproduction

https://stackblitz.com/edit/github-luiyyvqv?file=eslint.config.js

### System Info

```shell
System:
    OS: Windows 11 10.0.26100
    CPU: (12) x64 AMD Ryzen 5 5500
    Memory: 33.02 GB / 47.80 GB
  Binaries:
    Node: 23.11.0 - C:\Program Files\nodejs\node.EXE
    npm: 10.9.2 - C:\Program Files\nodejs\npm.CMD
    pnpm: 11.3.0 - C:\Users\flowers\AppData\Local\pnpm\bin\pnpm.CMD
    Deno: 2.9.4 - C:\Users\flowers\AppData\Local\Microsoft\WinGet\Packages\DenoLand.Deno_Microsoft.Winget.Source_8wekyb3d8bbwe\deno.EXE
  npmPackages:
    @stylistic/eslint-plugin: ^5.10.0 => 5.10.0
    eslint: ^9.39.5 => 9.39.5
```

### Contributes

- [ ] I am willing to submit a PR to fix this issue
- [ ] I am willing to submit a PR with failing tests
